File: cosmetic_mobile_app/lib/models/address.dart
class Address {
  final int id;
  final String title;
  final String city;
  final String district;
  final String fullAddress;

  Address({
    required this.id,
    required this.title,
    required this.city,
    required this.district,
    required this.fullAddress,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'],
      title: json['title'],
      city: json['city'],
      district: json['district'],
      fullAddress: json['fullAddress'],
    );
  }
}

File: src/main/java/com/cosmeticPlatform/CosmeticPlatform/model/Address.java
package com.cosmeticPlatform.CosmeticPlatform.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "address")
@Getter
@Setter
@com.fasterxml.jackson.annotation.JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title; // e.g., Home, Work

    private String city;

    private String district;

    @Column(length = 500)
    private String fullAddress;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;
}
